app-backend/db/src/test/scala/com/azavea/rf/database/TaskSessionDaoSpec.scala

```diff
@@ -437,4 +437,375 @@ class TaskSessionDaoSpec
     }
   }
 
+  test("Insert labels to session, toggle label, list label") {
+    check {
+      forAll {
+        (
+            userCreate: User.Create,
+            annotationProjectCreate: AnnotationProject.Create,
+            taskFeaturesCreate: Task.TaskFeatureCollectionCreate,
+            taskSessionCreate: TaskSession.Create,
+            annotationCreates: List[AnnotationLabelWithClasses.Create]
+        ) =>
+          {
+            val labelsOpt = annotationCreates.toNel
+            val listIO = for {
+              dbUser <- UserDao.create(userCreate)
+              dbAnnotationProj <-
+                AnnotationProjectDao
+                  .insert(
+                    annotationProjectCreate.copy(
+                      projectId = None,
+                      labelClassGroups =
+                        annotationProjectCreate.labelClassGroups.take(1)
+                    ),
+                    dbUser
+                  )
+              classIds = dbAnnotationProj.labelClassGroups flatMap {
+                _.labelClasses
+              } map { _.id }
+              fixedUpTasks = fixupTaskFeaturesCollection(
+                taskFeaturesCreate,
+                dbAnnotationProj,
+                None
+              )
+              task <- TaskDao.insertTasks(
```

Comment:
I think if we turn off all the foreign key constraints in tests, this IO could drop lines 453 to 471. Even with all the helpers it's just so much work. I'm feeling some regret about how all this works.

Reply:
@jisantuc yea, good call, though I have some small Qs about turning off foreign keys as I would not go as far as turning them _all_ off in tests. I am happy to discuss more in a call or in the issue/ADR for such a change as you brought up in scrum a few days ago.